src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,134 @@
+    List<Course> courses = getCourses(results);
+
+    String coursesJson = new Gson().toJson(courses);
+    response.setContentType("application/json;");
+    response.getWriter().println(coursesJson);
+  }
+
+  /* Create list of filters given parameters specified in request. */
+  public List<Filter> getFilters(HttpServletRequest request) {
+    List<Filter> filters = new ArrayList<>();
+    if (!request.getParameter("courseName").isEmpty()) {
+      String name = request.getParameter("courseName");
+      Filter nameFilter = new FilterPredicate("name", FilterOperator.EQUAL, name);
+      filters.add(nameFilter);
+    }
+
+    if (!request.getParameter("profName").isEmpty()) {
+      String profName = request.getParameter("profName");
+      Filter profFilter = new FilterPredicate("professor", FilterOperator.EQUAL, profName);
+      filters.add(profFilter);
+    }
+
+    if (!request.getParameter("term").equals("select")) {
+      String term = request.getParameter("term");
+      Filter termFilter = new FilterPredicate("term", FilterOperator.EQUAL, term);
+      filters.add(termFilter);
+    }
+
+    if (!request.getParameter("units").isEmpty()) {
+      List<Integer> units = new ArrayList<>();
+      List<String> strUnits = Arrays.asList(request.getParameter("units").split(","));
+      for (String number : strUnits) {
+        units.add(Integer.valueOf(number));
+      }
+      if (!units.isEmpty()) {
+        Filter unitsFilter = new FilterPredicate("units", FilterOperator.IN, units);
+        filters.add(unitsFilter);
+      }
+    }
+    return filters;
+  }
+
+  /* Combine filters, if applicable, and get results from Datastore matching this combination. */
+  public List<Entity> getResults(List<Filter> filters) {
+    Query courseQuery = new Query("Course");
+    if (!filters.isEmpty()) {
+      if (filters.size() == 1) {
+        courseQuery.setFilter(filters.get(0));
+      } else {
+        courseQuery.setFilter(CompositeFilterOperator.and(filters));
+      }
+    }
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    List<Entity> results =
+        datastore.prepare(courseQuery).asList(FetchOptions.Builder.withDefaults());
+    return results;
+  }
+
+  /* Given list of result Entity courses, format into Course objects. */
+  public List<Course> getCourses(List<Entity> results) {
+    List<Course> courses = new ArrayList<>();
+    for (Entity entity : results) {
+      String name = (String) entity.getProperty("name");
+      String professor = (String) entity.getProperty("professor");
+      Long numUnits = (Long) entity.getProperty("units");
+      String term = (String) entity.getProperty("term");
+      Course course = new Course(name, professor, numUnits, term);
+      courses.add(course);
+    }
+    return courses;
+  }
+
+  /* Store course. */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Ideally this should also be tested. That being said, since it's more for admins and not the end user I think it's alright to skip at this early stage. In a real world situation though you'd definitely want this to be tested too.

Reply:
Since we're planning on using Dagm's version of this and he's tested it, I figured it wasn't relevant - even if it's only used by me in this stage (and will be deleted after the merge/pull) should we still test ideally?

Reply:
Ah, in that case since this code will never be live in production then no need to worry I'd say